include/secp256k1_musig.h

```diff
@@ -375,113 +179,330 @@ SECP256K1_API int secp256k1_musig_partial_signature_serialize(
+ *       nonce_parity: the output of `musig_nonce_parity` called with the
+ *                     session used for producing sig64
  */
-SECP256K1_API int secp256k1_musig_partial_sig_adapt(
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_musig_adapt(
     const secp256k1_context* ctx,
-    secp256k1_musig_partial_signature *adaptor_sig,
-    const secp256k1_musig_partial_signature *partial_sig,
+    unsigned char *sig64,
     const unsigned char *sec_adaptor32,
     int nonce_parity
-) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
-/** Extracts a secret adaptor from a MuSig, given all parties' partial
- *  signatures. This function will not fail unless given grossly invalid data; if it
- *  is merely given signatures that do not verify, the returned value will be
- *  nonsense. It is therefore important that all data be verified at earlier steps of
- *  any protocol that uses this function.
- *
- *  Returns: 1: signatures contained valid data such that an adaptor could be extracted
- *           0: otherwise
- *  Args:         ctx: pointer to a context object (cannot be NULL)
- *  Out:sec_adaptor32: 32-byte secret adaptor (cannot be NULL)
- *  In:         sig64: complete 2-of-2 signature (cannot be NULL)
- *       partial_sigs: array of partial signatures (cannot be NULL)
- *     n_partial_sigs: number of elements in partial_sigs array
- *   nonce_parity: the `nonce_parity` output of `musig_session_combine_nonces`
+/** Extracts a secret adaptor from a MuSig pre-signature and corresponding
+ *  signature
+ *
+ *  This function will not fail unless given grossly invalid data; if it is
+ *  merely given signatures that do not verify, the returned value will be
+ *  nonsense. It is therefore important that all data be verified at earlier
+ *  steps of any protocol that uses this function. In particular, this includes
+ *  verifying all partial signatures that were aggregated into pre_sig64.
+ *
+ *  Returns: 0 if the arguments are NULL, or sig64 or pre_sig64 contain
+ *           grossly invalid (overflowing) values. 1 otherwise (which does NOT
+ *           mean the signatures or the adaptor are valid!)
+ *  Args:         ctx: pointer to a context object
+ *  Out:sec_adaptor32: 32-byte secret adaptor
+ *  In:         sig64: complete, valid 64-byte signature
+ *          pre_sig64: the pre-signature corresponding to sig64, i.e., the
+ *                     aggregate of partial signatures without the secret
+ *                     adaptor
+ *       nonce_parity: the output of `musig_nonce_parity` called with the
+ *                     session used for producing sig64
  */
-SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_musig_extract_secret_adaptor(
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_musig_extract_adaptor(
```

Comment:
Correct. This is because the adaptor sig module implements https://github.com/LLFourn/one-time-VES/blob/master/main.pdf which uses a different terminology.